http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/auth/SamAuthProvider.scala

```diff
@@ -142,13 +155,17 @@ class SamAuthProvider[F[_]: OpenTelemetryMetrics](
   ): F[List[(GoogleProject, R)]] = {
     val authHeader = Authorization(Credentials.Token(AuthScheme.Bearer, userInfo.accessToken.token))
     for {
-      projectPolicies <- samDao.getResourcePolicies[ProjectSamResourceId](authHeader)
-      owningProjects = projectPolicies.collect { case (r, SamPolicyName.Owner) =>
-        r.googleProject
-      }
-      resourcePolicies <- samDao
-        .getResourcePolicies[R](authHeader)
-      res = resourcePolicies.filter { case (_, pn) => sr.policyNames.contains(pn) }
+      projectPolicies <- resources.toList.flatTraverse(resource =>
```

Comment:
Yeah same as above -- unless I'm missing something it feels like this will result in much more Sam queries since we're calling `/api/resources/v2/{resourceType}` for each resource passed to this method. Maybe the code needs some restructuring…